### PR TITLE
Fixing up the metal arm/leg guards and adding more sheet metal armor.

### PR DESCRIPTION
**Pull request by Aerin-of-the-Toast**
#### Summary
Content "Sheet metal armor"

#### Purpose of change

The metal arm guards were interesting bits of sheet metal armor, but they were the *only* sheet metal armor (and quite encumbering for not touching the joints). This PR hopes to add a lot more sheet metal armor options, for the budding tank survivor who doesn't care too much about looks. Or comfort. Or moving anywhere.

#### Describe the solution

Swapped the material of the metal arm/leg guards to steel from iron, but halved the thickness to 1.5mm, which is the rough thickness of the in-game sheet metal. All other armor also uses this, so all the armor in this PR shares the same defensive values of 9 bash, 9 cut, and 4.5 ballistic. Also adjusted the proficiencies; joint articulation on something that explicitly avoids joints didn't make much sense, and armorsmithing required blacksmithing, so instead they all only require principles of metalworking instead.

Added sheet metal elbow and knee guards (low coverage, only 2 encumbrance since they don't restrict range of motion much at all), a sheet metal skirt for the hips and pauldrons for the shoulder (*much* higher encumbrance since they restrict range of motion significantly, low coverage in a vacuum but exist to cover the spots previously uncovered and make it much harder for enemy attacks to miss armor completely), and two variants of sheet metal chest guards. One that covers the whole torso but is significantly encumbering, and one that only covers the thorax but is far less encumbering.

Also added crafts for each of them. I used specified components instead of the generic cordage component under the assumption that string probably wouldn't hold sheet metal very well, so it instead uses the (criminally underused) leather lace exclusively, except the skirt and chest guards that use whole belts with the sheet metal riveted in with nuts & bolts or nails (since we don't have rivets). The crafting times were a bit of a shot in the dark though, so I did my best educated guess.

#### Describe alternatives you've considered

More sophisticated crafting recipes that might require i.e. sewing proper leather straps or the like, but I figured these were meant to be slapshod and made from scavenged junk.

#### Testing

Made sure everything spawned in game and the crafting recipes existed without getting hit with a JSON error.

#### Additional context

At some point I'd like to expand this to equally-slapshod sabatons, mitten gauntlets, and maybe some kind of helmet to eventually allow players to go full shitty sheet metal tank, if they wanted to.

### CODE REVIEW

File: data/json/items/armor/torso_armor.json [hunk 1]
```diff
@@ -225,6 +225,62 @@
...
+    "//": "The weight assumes the sheet metal is 75cm x 50cm x 1.5mm.",
+    "volume": "4 L",
+    "price": 500,
+    "price_postapoc": 400,
+    "to_hit": -2,
+    "material": [ "steel" ],
+    "symbol": "H",
+    "looks_like": "cuirass_lightplate",
+    "color": "light_gray",
+    "warmth": 10,
```
Comment (TheShadowFerret) [inline data/json/items/armor/torso_armor.json:244]: I'm pretty sure 1.5mm of *metal* has near-zero insulation value to it 🤔

(P.S. I think all of the existing pure-metal armors generally have too high warmth)

Comment (Aerin-of-the-Toast) [reply data/json/items/armor/torso_armor.json:244]: Yeah, technically they'd have negative insulation if that was possible, but I don't really think there's a way to make armor that sucks to wear in both winter and summer. I guess I could make it 0 warmth like chainmail and just let the 0 breathability represent that it'd still suck in summer.

Comment (Aerin-of-the-Toast) [reply data/json/items/armor/torso_armor.json:244]: Other pure-metal armors do have token warmth values though, and 10 is fairly low (lower than what they were previously at 20 warmth for the arm guards), so on review I'll probably just keep the warmths as-is. I'd love to fix it up if we ever get a proper insulation system though (though in that case it'll probably be handled automatically anyway).

**Comment by propurty** [comment]
Shouldn't metal of any substantial thickness be near impervious to cuts? Obviously from a game design standpoint that is kind've unbalanced, but irl metal is very resistant to cut. Bashing would be much more effective against metal armor. 

So because of that, wouldn't it make more sense to give any steel metal armor a very high cut protection but a comparably lower bash protection? I assume the sheet metal is made of steel or alloyed steel, which alloyed makes it stronger afaik. 

Medieval or knightly plate is usually between 1-3mm thick. Obviously through hammering and craftsmanship the metal is much thicker in some areas than others. Good armor was also angled to deflect, further increasing defense. 

Point being, if plate armors purpose was to completely nullify arrows and blades at only 1-3mm thick. Wouldn't modern day sheet metal do the same? Sure this is basic armor, not fine crafted. But it is modern steel and as thick as old plate armor.

Maybe I'm missing something important, just trying to start a discussion surrounding it. As the equivalent protection of bash and cut seemed off when I first saw it. 

Irl you would always want some kind of padding underneath your plate, which something like a gambeson could help spread the force of a blunt impact out. So maybe someone could work that into how effective layering and different types of armor are.

**Pull request merged**